identity_presentation/src/commonMain/kotlin/net/thechance/mena/identity/presentation/screen/changePassword/ChangePasswordScreenViewModel.kt

```diff
@@ -0,0 +1,169 @@
+            onError = ::onChangePasswordError
+        )
+
+    }
+
+    private suspend fun changePassword() {
+        val currentPassword = state.value.currentPasswordUIState.currentPassword
+        val newPassword = state.value.newPasswordUIState.newPassword
+        val confirmPassword = state.value.newPasswordUIState.confirmPassword
+
+        userRepository.changePassword(
+            currentPassword = currentPassword,
+            newPassword = newPassword,
+            confirmPassword = confirmPassword
+        )
+    }
+
+    private fun onChangePasswordSuccess() {
+        updateState { copy(isLoading = false) }
+        sendNewEffect(
+            ChangePasswordScreenUIEffect.NavigateBack(
+                snackBarUiState = SnackBarUiState(
+                    message = Res.string.changed_password_successfully,
+                    snackBarType = SnackBarType.SUCCESS,
+                    isVisible = true
+                )
+            )
+        )
+    }
+
+    private fun onChangePasswordError(throwable: Throwable) {
+        updateState { copy(errorMessage = mapErrorMessage(throwable), isLoading = false) }
+    }
+
+    override fun onChangeCurrentPassword(newValue: String) {
+        updateState {
+            copy(
+                currentPasswordUIState = currentPasswordUIState.copy(
+                    currentPassword = newValue
+                )
+            )
+        }
+        isContinueEnabled()
+    }
+
+    override fun onChangeNewPassword(newValue: String) {
+        updateState {
+            copy(
+                newPasswordUIState = newPasswordUIState.copy(
+                    newPassword = newValue
+                )
+            )
+        }
+        isSaveEnabled()
+    }
+
+    override fun onChangeConfirmPassword(newValue: String) {
+        updateState {
+            copy(
+                newPasswordUIState = newPasswordUIState.copy(
+                    confirmPassword = newValue
+                )
+            )
+        }
+        isSaveEnabled()
+    }
+
+    override fun onToggleCurrentPasswordVisibility() {
+        updateState {
+            copy(
+                currentPasswordUIState = currentPasswordUIState.copy(
+                    isCurrentPasswordVisible = !currentPasswordUIState.isCurrentPasswordVisible
+                )
+            )
+        }
+    }
+
+    override fun onToggleNewPasswordVisibility() {
+        updateState {
+            copy(
+                newPasswordUIState = newPasswordUIState.copy(
+                    isNewPasswordVisible = !newPasswordUIState.isNewPasswordVisible
+                )
+            )
+        }
+    }
+
+    override fun onToggleConfirmPasswordVisibility() {
+        updateState {
+            copy(
+                newPasswordUIState = newPasswordUIState.copy(
+                    isConfirmPasswordVisible = !newPasswordUIState.isConfirmPasswordVisible
+                )
+            )
+        }
+    }
+
+    private fun isSaveEnabled() {
```

Comment:
Renamed to clearly indicate that this function updates the state instead of returning a Boolean value this function actually update the state not return a Boolean value
```suggestion
    private fun updateSaveEnabledState() {
```